app.js

```diff
@@ -1,11 +1,26 @@
 const express = require("express");
 const app = express();
-const { getAllTopics, getApi } = require("./controllers/news.controller");
+const {
+  getAllTopics,
+  getApi,
+  getArticleByArticleId,
+} = require("./controllers/news.controller");
 const { notFoundHandler } = require("./error_handling/errors");
 
 app.get("/api/topics", getAllTopics);
 
 app.get("/api", getApi);
+
+app.get("/api/articles/:article_id", getArticleByArticleId);
+
+app.use((err, request, response, next) => {
+  if (err.code === "22P02") {
+    response.status(400).send({ message: "Bad request" });
+  } else if (response) {
+    response.status(404).send({ message: "Not found" });
+  } else next();
+});
+
 // CODE BELOW IS NOT WORKING
 
 app.get("/api/*", function (req, res) {
```

Comment:
This isn't working because app.get needs to be above any app.use. If you move this to above your app.use for error handling it will work.

Also look at the error handling notes again, specifically at the part where each error handler is extracted into its own function. It is very specific and gives you a step by step on how to do it.